Mathlib/Algebra/Algebra/Quasispectrum.lean

```diff
@@ -0,0 +1,291 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Algebra.Algebra.Unitization
+import Mathlib.Algebra.Algebra.Spectrum
+
+/-!
+# Quasiregularity and quasispectrum
+
+For a non-unital ring `R`, an element `r : R` is *quasiregular* if it is invertible in the monoid
+`(R, ∘)` where `x ∘ y := y + x + x * y` with identity `0 : R`. We implement this both as a type
+synonym `Quasiregular` which has an associated `Monoid` instance (note: *not* an `AddMonoid`
+instance despite the fact that `0 : R` is the identity in this monoid) so that one may access
+the quasiregular elements of `R` as `(Quasiregular R)ˣ`, but also as a predicate `IsQuasiregular`.
+
+Quasiregularity is closely tied to invertibility. Indeed, `(Quasiregular A)ˣ` is isomorphic to the
+subgroup of `Unitization R A` whose scalar part is `1`, whenever `A` is a non-unital `R`-algebra,
+and moreover this isomorphism is implemented by the map `(x : A) ↦ (1 + x : Unitization R A)`. It
+is because of this isomorphism, and the assoicated ties with multiplicative invertibility, that we
+choose a `Monoid` (as opposed to an `AddMonoid`) structure on `Quasiregular`.  In addition,
```

Comment:
```suggestion
is because of this isomorphism, and the associated ties with multiplicative invertibility, that we
choose a `Monoid` (as opposed to an `AddMonoid`) structure on `Quasiregular`.  In addition,
```